Cannot remove all items from single collection or checklist

See http://localhost:8080/content/admin/objects.html/path:/content/objects/example

Try to remove all of the items in the single collection and click okay.
The payload sent to the server shows the items successfully removed, but are re added in the response.

@reusr1 

**trying to manually reproduce it I was able to consistently hit a HTTP-500 error**

## Reproduction Steps

- open tenant `example site` (internal websites)
- move to objects page
- create new object based on `collection`
- fill some values into the 2 collections in the wizard
- click finish
- edit the newly created object and remove all items from either collection
- press save
- **HTTP-500**

## Response
```prolog
{
  "code" : -124,
  "message" : "Failed with IO exception",
  "exception" : [
    "org.apache.sling.api.resource.PersistenceException: Unable to commit changes to session.",
    "\tat org.apache.sling.jcr.resource.internal.helper.jcr.JcrResourceProvider.commit(JcrResourceProvider.java:516)",
    "\tat org.apache.sling.resourceresolver.impl.providers.stateful.AuthenticatedResourceProvider.commit(AuthenticatedResourceProvider.java:215)",
    "\tat org.apache.sling.resourceresolver.impl.helper.ResourceResolverControl.commit(ResourceResolverControl.java:424)",
    "\tat org.apache.sling.resourceresolver.impl.ResourceResolverImpl.commit(ResourceResolverImpl.java:997)",
    "\tat com.peregrine.admin.servlets.UpdateResourceServlet.handleRequest(UpdateResourceServlet.java:94)",
    "\tat com.peregrine.commons.servlets.AbstractBaseServlet.doRequest(AbstractBaseServlet.java:90)",
    "\tat com.peregrine.commons.servlets.AbstractBaseServlet.doPost(AbstractBaseServlet.java:84)",
    "\tat org.apache.sling.api.servlets.SlingAllMethodsServlet.mayService(SlingAllMethodsServlet.java:146)",
    "\tat org.apache.sling.api.servlets.SlingSafeMethodsServlet.service(SlingSafeMethodsServlet.java:342)",
    "\tat org.apache.sling.api.servlets.SlingSafeMethodsServlet.service(SlingSafeMethodsServlet.java:374)",
    "\tat org.apache.sling.engine.impl.request.RequestData.service(RequestData.java:552)",
    "\tat org.apache.sling.engine.impl.filter.SlingComponentFilterChain.render(SlingComponentFilterChain.java:44)",
    "\tat org.apache.sling.engine.impl.filter.AbstractSlingFilterChain.doFilter(AbstractSlingFilterChain.java:82)",
    "\tat org.apache.sling.engine.impl.SlingRequestProcessorImpl.processComponent(SlingRequestProcessorImpl.java:283)",
    "\tat org.apache.sling.engine.impl.filter.RequestSlingFilterChain.render(RequestSlingFilterChain.java:49)",
    "\tat org.apache.sling.engine.impl.filter.AbstractSlingFilterChain.doFilter(AbstractSlingFilterChain.java:82)",
    "\tat com.peregrine.versions.VersioningRequestFilter.doFilter(VersioningRequestFilter.java:51)",
    "\tat org.apache.sling.engine.impl.filter.AbstractSlingFilterChain.doFilter(AbstractSlingFilterChain.java:72)",
    "\tat org.apache.sling.engine.impl.debug.RequestProgressTrackerLogFilter.doFilter(RequestProgressTrackerLogFilter.java:110)",
    "\tat org.apache.sling.engine.impl.filter.AbstractSlingFilterChain.doFilter(AbstractSlingFilterChain.java:72)",
    "\tat org.apache.sling.i18n.impl.I18NFilter.doFilter(I18NFilter.java:131)",
    "\tat org.apache.sling.engine.impl.filter.AbstractSlingFilterChain.doFilter(AbstractSlingFilterChain.java:72)",
    "\tat org.apache.sling.engine.impl.filter.AbstractSlingFilterChain.doFilter(AbstractSlingFilterChain.java:78)",
    "\tat org.apache.sling.engine.impl.SlingRequestProcessorImpl.doProcessRequest(SlingRequestProcessorImpl.java:151)",
    "\tat org.apache.sling.engine.impl.SlingMainServlet.service(SlingMainServlet.java:250)",
    "\tat org.apache.felix.http.base.internal.handler.ServletHandler.handle(ServletHandler.java:123)",
    "\tat org.apache.felix.http.base.internal.dispatch.InvocationChain.doFilter(InvocationChain.java:86)",
    "\tat org.apache.sling.junit.impl.servlet.TestLogServlet$TestNameLoggingFilter.doFilter(TestLogServlet.java:257)",
    "\tat org.apache.felix.http.base.internal.handler.FilterHandler.handle(FilterHandler.java:142)",
    "\tat org.apache.felix.http.base.internal.dispatch.InvocationChain.doFilter(InvocationChain.java:81)",
    "\tat org.apache.sling.i18n.impl.I18NFilter.doFilter(I18NFilter.java:131)",
    "\tat org.apache.felix.http.base.internal.handler.FilterHandler.handle(FilterHandler.java:142)",
    "\tat org.apache.felix.http.base.internal.dispatch.InvocationChain.doFilter(InvocationChain.java:81)",
    "\tat org.apache.sling.engine.impl.log.RequestLoggerFilter.doFilter(RequestLoggerFilter.java:75)",
    "\tat org.apache.felix.http.base.internal.handler.FilterHandler.handle(FilterHandler.java:142)",
    "\tat org.apache.felix.http.base.internal.dispatch.InvocationChain.doFilter(InvocationChain.java:81)",
    "\tat org.apache.sling.engine.impl.parameters.RequestParameterSupportConfigurer.doFilter(RequestParameterSupportConfigurer.java:67)",
    "\tat org.apache.felix.http.base.internal.handler.FilterHandler.handle(FilterHandler.java:142)",
    "\tat org.apache.felix.http.base.internal.dispatch.InvocationChain.doFilter(InvocationChain.java:81)",
    "\tat org.apache.felix.http.base.internal.dispatch.Dispatcher$1.doFilter(Dispatcher.java:146)",
    "\tat org.apache.felix.http.base.internal.whiteboard.WhiteboardManager$2.doFilter(WhiteboardManager.java:1002)",
    "\tat org.apache.felix.http.sslfilter.internal.SslFilter.doFilter(SslFilter.java:97)",
    "\tat org.apache.felix.http.base.internal.handler.PreprocessorHandler.handle(PreprocessorHandler.java:136)",
    "\tat org.apache.felix.http.base.internal.whiteboard.WhiteboardManager$2.doFilter(WhiteboardManager.java:1008)",
    "\tat org.apache.felix.http.base.internal.whiteboard.WhiteboardManager.invokePreprocessors(WhiteboardManager.java:1012)",
    "\tat org.apache.felix.http.base.internal.dispatch.Dispatcher.dispatch(Dispatcher.java:91)",
    "\tat org.apache.felix.http.base.internal.dispatch.DispatcherServlet.service(DispatcherServlet.java:49)",
    "\tat javax.servlet.http.HttpServlet.service(HttpServlet.java:725)",
    "\tat org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:763)",
    "\tat org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:551)",
    "\tat org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233)",
    "\tat org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1610)",
    "\tat org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233)",
    "\tat org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1363)",
    "\tat org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188)",
    "\tat org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:489)",
    "\tat org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1580)",
    "\tat org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186)",
    "\tat org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1278)",
    "\tat org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)",
    "\tat org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:221)",
    "\tat org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)",
    "\tat org.eclipse.jetty.server.Server.handle(Server.java:500)",
    "\tat org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:383)",
    "\tat org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:547)",
    "\tat org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:375)",
    "\tat org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:273)",
    "\tat org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)",
    "\tat org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)",
    "\tat org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:117)",
    "\tat org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:336)",
    "\tat org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:313)",
    "\tat org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:171)",
    "\tat org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:129)",
    "\tat org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:375)",
    "\tat org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:806)",
    "\tat org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:938)",
    "\tat java.base/java.lang.Thread.run(Thread.java:834)",
    "Caused by: javax.jcr.InvalidItemStateException: OakState0001: Unresolved conflicts in /content/example/objects/my-collection",
    "\tat org.apache.jackrabbit.oak.api.CommitFailedException.asRepositoryException(CommitFailedException.java:238)",
    "\tat org.apache.jackrabbit.oak.api.CommitFailedException.asRepositoryException(CommitFailedException.java:213)",
    "\tat org.apache.jackrabbit.oak.jcr.delegate.SessionDelegate.newRepositoryException(SessionDelegate.java:669)",
    "\tat org.apache.jackrabbit.oak.jcr.delegate.SessionDelegate.save(SessionDelegate.java:495)",
    "\tat org.apache.jackrabbit.oak.jcr.session.SessionImpl$8.performVoid(SessionImpl.java:424)",
    "\tat org.apache.jackrabbit.oak.jcr.delegate.SessionDelegate.performVoid(SessionDelegate.java:273)",
    "\tat org.apache.jackrabbit.oak.jcr.session.SessionImpl.save(SessionImpl.java:421)",
    "\tat org.apache.sling.jcr.oak.server.internal.TcclWrappingJackrabbitSession.save(TcclWrappingJackrabbitSession.java:212)",
    "\tat org.apache.sling.jcr.resource.internal.helper.jcr.JcrResourceProvider.commit(JcrResourceProvider.java:514)",
    "\t... 77 more",
    "Caused by: org.apache.jackrabbit.oak.api.CommitFailedException: OakState0001: Unresolved conflicts in /content/example/objects/my-collection",
    "\tat org.apache.jackrabbit.oak.plugins.commit.ConflictValidator.failOnMergeConflict(ConflictValidator.java:115)",
    "\tat org.apache.jackrabbit.oak.plugins.commit.ConflictValidator.propertyAdded(ConflictValidator.java:84)",
    "\tat org.apache.jackrabbit.oak.spi.commit.CompositeEditor.propertyAdded(CompositeEditor.java:82)",
    "\tat org.apache.jackrabbit.oak.spi.commit.EditorDiff.propertyAdded(EditorDiff.java:81)",
    "\tat org.apache.jackrabbit.oak.segment.SegmentNodeState.compareProperties(SegmentNodeState.java:660)",
    "\tat org.apache.jackrabbit.oak.segment.SegmentNodeState.compareAgainstBaseState(SegmentNodeState.java:523)",
    "\tat org.apache.jackrabbit.oak.spi.commit.EditorDiff.childNodeChanged(EditorDiff.java:147)",
    "\tat org.apache.jackrabbit.oak.segment.MapRecord.compare(MapRecord.java:444)",
    "\tat org.apache.jackrabbit.oak.segment.SegmentNodeState.compareAgainstBaseState(SegmentNodeState.java:651)",
    "\tat org.apache.jackrabbit.oak.spi.commit.EditorDiff.childNodeChanged(EditorDiff.java:147)",
    "\tat org.apache.jackrabbit.oak.segment.MapRecord.compare(MapRecord.java:444)",
    "\tat org.apache.jackrabbit.oak.segment.SegmentNodeState.compareAgainstBaseState(SegmentNodeState.java:651)",
    "\tat org.apache.jackrabbit.oak.spi.commit.EditorDiff.childNodeChanged(EditorDiff.java:147)",
    "\tat org.apache.jackrabbit.oak.segment.MapRecord.compare(MapRecord.java:444)",
    "\tat org.apache.jackrabbit.oak.segment.SegmentNodeState.compareAgainstBaseState(SegmentNodeState.java:651)",
    "\tat org.apache.jackrabbit.oak.spi.commit.EditorDiff.childNodeChanged(EditorDiff.java:147)",
    "\tat org.apache.jackrabbit.oak.segment.MapRecord.compare(MapRecord.java:444)",
    "\tat org.apache.jackrabbit.oak.segment.SegmentNodeState.compareAgainstBaseState(SegmentNodeState.java:651)",
    "\tat org.apache.jackrabbit.oak.spi.commit.EditorDiff.process(EditorDiff.java:51)",
    "\tat org.apache.jackrabbit.oak.spi.commit.EditorHook.processCommit(EditorHook.java:54)",
    "\tat org.apache.jackrabbit.oak.spi.commit.CompositeHook.processCommit(CompositeHook.java:60)",
    "\tat org.apache.jackrabbit.oak.spi.commit.CompositeHook.processCommit(CompositeHook.java:60)",
    "\tat org.apache.jackrabbit.oak.segment.scheduler.Commit.apply(Commit.java:105)",
    "\tat org.apache.jackrabbit.oak.segment.scheduler.LockBasedScheduler.execute(LockBasedScheduler.java:299)",
    "\tat org.apache.jackrabbit.oak.segment.scheduler.LockBasedScheduler.schedule(LockBasedScheduler.java:270)",
    "\tat org.apache.jackrabbit.oak.segment.SegmentNodeStore.merge(SegmentNodeStore.java:211)",
    "\tat org.apache.jackrabbit.oak.core.MutableRoot.commit(MutableRoot.java:251)",
    "\tat org.apache.jackrabbit.oak.jcr.delegate.SessionDelegate.commit(SessionDelegate.java:346)",
    "\tat org.apache.jackrabbit.oak.jcr.delegate.SessionDelegate.save(SessionDelegate.java:493)",
    "\t... 82 more"
  ]
}
```

--- a/admin-base/ui.apps/src/main/content/jcr_root/apps/admin/components/explorer/template.vue
+++ b/admin-base/ui.apps/src/main/content/jcr_root/apps/admin/components/explorer/template.vue
@@ -534,8 +534,9 @@ export default {
             showInfo: function(me, target) {
                 const tenant = $perAdminApp.getView().state.tenant
                 if(target.startsWith(`/content/${tenant.name}/objects`)) {
-                    const node = $perAdminApp.findNodeFromPath($perAdminApp.getView().admin.nodes, target)
-                    $perAdminApp.stateAction('selectObject', { selected: node.path, path: me.model.dataFrom })
+                  const node = $perAdminApp.findNodeFromPath($perAdminApp.getView().admin.nodes, target)
+                  set($perAdminApp.getView(), `/state/tools/edit`, false)
+                  $perAdminApp.stateAction('selectObject', { selected: node.path, path: me.model.dataFrom })
                 } else if (target.startsWith(`/content/${tenant.name}/templates`)) {
                     $perAdminApp.stateAction('showTemplateInfo', { selected: target })
                 } else {

--- a/admin-base/ui.apps/src/main/content/jcr_root/apps/admin/components/explorerpreviewcontent/template.vue
+++ b/admin-base/ui.apps/src/main/content/jcr_root/apps/admin/components/explorerpreviewcontent/template.vue
@@ -542,6 +542,9 @@ export default {
       if (this.stateToolsEdit) {
         this.onEdit()
       }
+    },
+    stateToolsEdit(edit) {
+      this.edit = edit
     }
   },
   created() {

--- a/admin-base/ui.apps/src/main/content/jcr_root/apps/admin/components/explorerpreviewcontent/template.vue
+++ b/admin-base/ui.apps/src/main/content/jcr_root/apps/admin/components/explorerpreviewcontent/template.vue
@@ -306,7 +306,7 @@
 
 <script>
 import {IconLib, MimeType, NodeType, SUFFIX_PARAM_SEPARATOR} from '../../../../../../js/constants'
-import {deepClone, get} from '../../../../../../js/utils'
+import {deepClone, get, set} from '../../../../../../js/utils'
 import NodeNameValidation from '../../../../../../js/mixins/NodeNameValidation'
 import ReferenceUtil from '../../../../../../js/mixins/ReferenceUtil'
 import Icon from '../icon/template.vue'
@@ -851,6 +851,7 @@ export default {
           data[key] = targetNode;
         }
       }
+      set($perAdminApp.getView(), '/state/tools/save/confirmed', true)
       $perAdminApp.stateAction('saveObjectEdit', {data: data, path: show}).then(() => {
         $perAdminApp.getNodeFromView('/state/tools')._deleted = {}
       });

--- a/admin-base/ui.apps/src/main/content/jcr_root/apps/admin/components/explorerpreviewcontent/template.vue
+++ b/admin-base/ui.apps/src/main/content/jcr_root/apps/admin/components/explorerpreviewcontent/template.vue
@@ -507,11 +507,19 @@ export default {
     },
     activationSensitiveClass() {
       return this.selfOrAnyDescendantActivated ? 'operationDisabledOnActivatedItem' : null;
+    },
+    stateToolsEdit() {
+      const stateTools = $perAdminApp.getNodeFromViewOrNull('/state/tools')
+      if (stateTools) {
+        return stateTools.edit
+      } else {
+        return false
+      }
     }
   },
   watch: {
-    edit(newVal) {
-      $perAdminApp.getNodeFromView('/state/tools').edit = newVal;
+    edit(val) {
+      $perAdminApp.getNodeFromViewOrNull('/state/tools').edit = val
     },
     activeTab : function(tab) {
       if (tab === 'versions'){
@@ -525,6 +533,9 @@ export default {
         this.showVersions()
       } else if (this.activeTab === 'references'){
         this.showReferencedBy()
+      }
+      if (this.stateToolsEdit) {
+        this.onEdit()
       }
     }
   },
@@ -621,6 +632,10 @@ export default {
     onEdit() {
       this.edit = true
       this.formGenerator.original = deepClone(this.node)
+
+      if (this.nodeType === NodeType.OBJECT) {
+        $perAdminApp.stateAction('editObject', {selected: this.currentObject})
+      }
     },
     onCancel() {
       const payload = {selected: this.currentObject}

--- a/admin-base/ui.apps/src/main/content/jcr_root/apps/field/collection/template.vue
+++ b/admin-base/ui.apps/src/main/content/jcr_root/apps/field/collection/template.vue
@@ -148,6 +148,9 @@
             this.prepModel(newChild, this.schema)
             newChild['sling:resourceType'] = this.schema.resourceType
         }
+        if (!this.value) {
+          this.value = []
+        }
         this.value.push(newChild)
         // Vue.set(this.value, this.value.length -1, newChild)
         this.onSetActiveItem(this.value.length - 1)

--- a/admin-base/ui.apps/src/main/content/jcr_root/apps/field/collection/template.vue
+++ b/admin-base/ui.apps/src/main/content/jcr_root/apps/field/collection/template.vue
@@ -27,9 +27,10 @@
     <label>{{schema.title}} </label>
     <ul v-if="!schema.preview" class="collapsible" v-bind:class="schema.multifield ? 'multifield' : 'singlefield'" ref="collapsible">
         <li v-for="(item, index) in value" v-bind:class="getItemClass(item, index)" v-bind:key="item.path"> {{item._opDelete}}
-            <div 
-              class="collapsible-header" 
-              draggable="true" 
+            <div
+              ref="header"
+              class="collapsible-header"
+              draggable="true"
               v-on:dragstart = "onDragStart(item, index, $event)"
               v-on:dragover.prevent  ="onDragOver"
               v-on:dragenter.prevent ="onDragEnter"
@@ -37,9 +38,10 @@
               v-on:drop.prevent      ="onDrop(item, index, $event)"
               v-on:click.stop.prevent="onSetActiveItem(index)">
                 <i class="material-icons">drag_handle</i>
-                <span v-if="schema.multifield">{{itemName(item, index)}}</span> 
+                <span v-if="schema.multifield">{{itemName(item, index)}}</span>
                 <input
                   v-else
+                  ref="input"
                   v-model="value[index]">
                 <i class="material-icons delete-icon" v-on:click="onRemoveItem(item, index)">delete</i>
             </div>
@@ -63,7 +65,7 @@
 
         <ul v-if="typeof item === 'object'" class="collection z-depth-1" :key="item.name">
           <vue-form-generator
-            v-if="schema.multifield" 
+            v-if="schema.multifield"
             class="collection-item"
             :schema="schema"
             :model="prepModel(item, schema)"></vue-form-generator>
@@ -170,30 +172,37 @@
         }
       },
       onSetActiveItem(index){
-        if(!this.schema.multifield) return
-        if(index === this.activeItem){
-          $(this.$refs.collapsible).collapsible('close', this.activeItem)
-          this.activeItem = null
-        } else {
-          this.$nextTick(function () {
-            if(this.activeItem !== null){
-              $(this.$refs.collapsible).collapsible('close', this.activeItem)
+        if(!this.schema.multifield) {
+          this.$nextTick(() => {
+            if (this.$refs.input[index]) {
+              this.$refs.input[index].focus()
             }
-            $(this.$refs.collapsible).collapsible('open', index)
-            this.activeItem = index
-            // focus first field of expanded item
-            this.$nextTick(() => {
-              let firstField = this.$refs.collapsible.querySelector('li.active input')
-              if(firstField) firstField.focus()
-            })
           })
+        } else {
+          if(index === this.activeItem){
+            $(this.$refs.collapsible).collapsible('close', this.activeItem)
+            this.activeItem = null
+          } else {
+            this.$nextTick(function () {
+              if(this.activeItem !== null){
+                $(this.$refs.collapsible).collapsible('close', this.activeItem)
+              }
+              $(this.$refs.collapsible).collapsible('open', index)
+              this.activeItem = index
+              // focus first field of expanded item
+              this.$nextTick(() => {
+                let firstField = this.$refs.collapsible.querySelector('li.active input')
+                if(firstField) firstField.focus()
+              })
+            })
+          }
         }
       },
       onDragStart(item, index, ev){
         ev.dataTransfer.setData('text', index)
       },
       onDragOver(ev){
-        const center = ev.target.offsetHeight / 2 
+        const center = ev.target.offsetHeight / 2
         ev.target.classList.toggle('drop-after', ev.offsetY > center )
         ev.target.classList.toggle('drop-before', ev.offsetY < center )
       },

--- a/admin-base/ui.apps/src/main/js/stateActions/createObject.js
+++ b/admin-base/ui.apps/src/main/js/stateActions/createObject.js
@@ -36,6 +36,7 @@ export default function(me, target) {
                 api.saveObjectEdit(target.parent + '/' + target.name, target.data).then( () => {
                     if(target.returnTo) {
                         me.loadContent(target.returnTo+'.html/path' +SUFFIX_PARAM_SEPARATOR + target.parent)
+                        me.stateAction('selectObject', { selected: target.parent + '/' + target.name, path: '/state/tools/objects' })
                     } else {
                         me.loadContent('/content/admin/pages/objects.html/path' + SUFFIX_PARAM_SEPARATOR + target.parent)
                     }

--- a/admin-base/ui.apps/src/main/js/stateActions/editObject.js
+++ b/admin-base/ui.apps/src/main/js/stateActions/editObject.js
@@ -23,7 +23,7 @@
  * #L%
  */
 import {LoggerFactory} from '../logger'
-import {deepClone, set} from '../utils'
+import {deepClone, get, set} from '../utils'
 
 let log = LoggerFactory.logger('editObject').setLevelDebug()
 
@@ -35,6 +35,8 @@ export default function(me, target) {
     set(me.getView(), '/state/tools/save/confirmed', false)
 
     me.beforeStateAction((name) => {
+        const confirmed = get(me.getView(), '/state/tools/save/confirmed',
+            false)
         const currentObject = deepClone(
             me.getNodeFromView('/state/tools/object'))
 
@@ -42,7 +44,7 @@ export default function(me, target) {
             // if there was no change skip asking to save
             const newChecksum = JSON.stringify(
                 me.getNodeFromView('/state/tools/object/data'))
-            if (checksum === newChecksum) {
+            if (confirmed || checksum === newChecksum) {
                 return true
             } else {
                 return new Promise((resolve) => {

--- a/admin-base/ui.apps/src/main/js/stateActions/editObject.js
+++ b/admin-base/ui.apps/src/main/js/stateActions/editObject.js
@@ -34,31 +34,39 @@ export default function(me, target) {
     let checksum = ''
     set(me.getView(), '/state/tools/save/confirmed', false)
 
-    me.beforeStateAction( (name) => {
-        const confirmed = get(me.getView(), '/state/tools/save/confirmed', false)
-        const currentObject = deepClone(me.getNodeFromView('/state/tools/object'))
+    me.beforeStateAction((name) => {
+        const confirmed = get(me.getView(), '/state/tools/save/confirmed',
+            false)
+        const currentObject = deepClone(
+            me.getNodeFromView('/state/tools/object'))
 
-        if(name !== 'saveObjectEdit') {
+        if (name !== 'saveObjectEdit') {
             // if there was no change skip asking to save
-            const newChecksum = JSON.stringify(me.getNodeFromView('/state/tools/object/data'))
-            if(confirmed || checksum === newChecksum) {
+            const newChecksum = JSON.stringify(
+                me.getNodeFromView('/state/tools/object/data'))
+            if (confirmed || checksum === newChecksum) {
                 return true
             } else {
-                $perAdminApp.askUser(
-                    'Save Object Edit?',
-                    'Would you like to save your object edits?',
-                    {
-                        yesText: 'Save',
-                        noText: 'Cancel',
-                        yes() {
-                            me.stateAction('saveObjectEdit', {
-                                data: currentObject.data,
-                                path: currentObject.show
-                            })
-                        },
-                        no: () => {}
-                    }
-                )
+                return new Promise((resolve) => {
+                    $perAdminApp.askUser(
+                        'Save Object Edit?',
+                        'Would you like to save your object edits?',
+                        {
+                            yesText: 'Save',
+                            noText: 'Cancel',
+                            yes() {
+                                me.stateAction('saveObjectEdit', {
+                                    data: currentObject.data,
+                                    path: currentObject.show
+                                })
+                                resolve()
+                            },
+                            no() {
+                                resolve()
+                            }
+                        }
+                    )
+                })
             }
         }
         return true

--- a/admin-base/ui.apps/src/main/js/stateActions/editObject.js
+++ b/admin-base/ui.apps/src/main/js/stateActions/editObject.js
@@ -23,7 +23,7 @@
  * #L%
  */
 import {LoggerFactory} from '../logger'
-import {deepClone, set, get} from '../utils'
+import {deepClone, set} from '../utils'
 
 let log = LoggerFactory.logger('editObject').setLevelDebug()
 
@@ -35,8 +35,6 @@ export default function(me, target) {
     set(me.getView(), '/state/tools/save/confirmed', false)
 
     me.beforeStateAction((name) => {
-        const confirmed = get(me.getView(), '/state/tools/save/confirmed',
-            false)
         const currentObject = deepClone(
             me.getNodeFromView('/state/tools/object'))
 
@@ -44,7 +42,7 @@ export default function(me, target) {
             // if there was no change skip asking to save
             const newChecksum = JSON.stringify(
                 me.getNodeFromView('/state/tools/object/data'))
-            if (confirmed || checksum === newChecksum) {
+            if (checksum === newChecksum) {
                 return true
             } else {
                 return new Promise((resolve) => {
@@ -73,9 +71,10 @@ export default function(me, target) {
     })
 
     let view = me.getView()
-    Vue.set(me.getNodeFromView('/state/tools'), 'edit', true)
+    set(me.getView(), `/state/tools/edit`, false)
     me.getApi().populateObject(target.selected, '/state/tools/object', 'data').then( () => {
         checksum = JSON.stringify(me.getNodeFromView('/state/tools/object/data'))
         set(view, '/state/tools/object/show', target.selected)
+        set(me.getView(), `/state/tools/edit`, true)
     })
 }

--- a/admin-base/ui.apps/src/main/js/stateActions/editObject.js
+++ b/admin-base/ui.apps/src/main/js/stateActions/editObject.js
@@ -22,27 +22,43 @@
  * under the License.
  * #L%
  */
-import { LoggerFactory } from '../logger'
-let log = LoggerFactory.logger('editObject').setLevelDebug()
+import {LoggerFactory} from '../logger'
+import {deepClone, set, get} from '../utils'
 
-import { set } from '../utils'
+let log = LoggerFactory.logger('editObject').setLevelDebug()
 
 export default function(me, target) {
 
     log.fine(target)
 
     let checksum = ''
+    set(me.getView(), '/state/tools/save/confirmed', false)
 
-    me.beforeStateAction( function(name) {
+    me.beforeStateAction( (name) => {
+        const confirmed = get(me.getView(), '/state/tools/save/confirmed', false)
+        const currentObject = deepClone(me.getNodeFromView('/state/tools/object'))
+
         if(name !== 'saveObjectEdit') {
             // if there was no change skip asking to save
-            if(checksum === JSON.stringify(me.getNodeFromView('/state/tools/object/data'))) {
+            const newChecksum = JSON.stringify(me.getNodeFromView('/state/tools/object/data'))
+            if(confirmed || checksum === newChecksum) {
                 return true
-            }
-            const yes = confirm('save edit?')
-            if(yes) {
-                const currentObject = me.getNodeFromView("/state/tools/object")
-                me.stateAction('saveObjectEdit', { data: currentObject.data, path: currentObject.show })
+            } else {
+                $perAdminApp.askUser(
+                    'Save Object Edit?',
+                    'Would you like to save your object edits?',
+                    {
+                        yesText: 'Save',
+                        noText: 'Cancel',
+                        yes() {
+                            me.stateAction('saveObjectEdit', {
+                                data: currentObject.data,
+                                path: currentObject.show
+                            })
+                        },
+                        no: () => {}
+                    }
+                )
             }
         }
         return true

--- a/admin-base/ui.apps/src/main/js/stateActions/saveObjectEdit.js
+++ b/admin-base/ui.apps/src/main/js/stateActions/saveObjectEdit.js
@@ -23,12 +23,14 @@
  * #L%
  */
 import { LoggerFactory } from '../logger'
+import {set} from '../utils'
 let log = LoggerFactory.logger('saveObjectEdit').setLevelDebug()
 
 export default function(me, target) {
 
     log.fine(target)
 
+    set(me.getView(), '/state/tools/save/confirmed', false)
     return new Promise( (resolve, reject) => {
         me.getApi().saveObjectEdit(target.path, target.data).then( () => {
             resolve()

--- a/admin-base/ui.apps/src/main/js/stateActions/selectObject.js
+++ b/admin-base/ui.apps/src/main/js/stateActions/selectObject.js
@@ -34,7 +34,7 @@ export default function(me, target) {
     let view = me.getView()
 
     return new Promise( (resolve, reject) => {
-        Vue.set(me.getNodeFromView('/state/tools'), 'edit', false)
+        set(view, '/state/tools/edit', false)
         me.getApi().populateObject(target.selected, '/state/tools/object', 'data')
         .then( () => {
             set(view, '/state/tools/object/show', target.selected);

